Create Hero Section

Warri0r2005 opened the pull request:
closes #72 

## 🌟 Context <!-- What is the purpose of this PR? -->
- We want a functioning hero page to help viewers join the club
<!--
Link to related issues, user stories, or tasks: #<issue_number>
Mention relevant sprints or milestones.
Describe the main goal of this change and why it's needed
Include any business context or requirements driving this change
-->

---

## 📝 Description <!--What changes have been made? -->
- Updated stylings for images and text
- Updated CTA button styling
<!--
Briefly describe key updates — new features, bug fixes, or refactors.
Highlight any important design or architectural decisions.
Explain how these changes address the issue/requirement
Include any dependencies that are required for this change
-->

---

## 📋 Notes <!--Are there any important details for reviewers? -->
- No animation - Not sure what to go off.
<!--
Potential risks or breaking changes.
Edge cases or areas requiring special attention.
Anything you want feedback on!
List any manual testing you've performed
Mention any documentation that needs updating as a result
-->

phawatboom commented:
Nice! Maybe can make this equal? 
![image](https://github.com/user-attachments/assets/ef05dff1-1651-43e2-bd65-541786d820f2)

dyzhuu reviewed (changes requested):
Good start! There’s still a lot of inconsistency between this and the figma design though, since the mobile design is complete you can start with the spacing and font sizes in that design.

No need for animations at this stage :)

Warri0r2005 commented on src/app/(frontend)/_components/sections/HomePage/Hero/BackgroundImage.tsx:
```diff
@@ -7,11 +7,14 @@ interface BackgroundImageProps {
+        src={src}
+        alt={alt}
+        fill
+        className="absolute inset-0 z-0 h-full w-full translate-y-[-60px] scale-130 object-cover brightness-90 md:translate-x-25 md:scale-120"
```
Yes its to fit the figma design, up to you wether you want to crop it or not

dyzhuu replied:
Ok, I think don't transform the image with css, we can crop the source image instead.

dyzhuu commented:
<img width="400" alt="image" src="https://github.com/user-attachments/assets/e6bec37e-47f8-4ac5-8124-ceffe45c8460" />
<img width="400" alt="image" src="https://github.com/user-attachments/assets/091d8481-3eef-4206-8936-c823af51309a" />


There are still some discrepancies between this and the figma design. I would double check
* the font sizes between the mobile, tablet, and figma designs, and how you have it (currently the figma specifies 16px for desktop but you have it as 20px)
* the "welcome to AUCC" text also seems bigger 
* there also seems to be a light gradient overlay which you don't have


<img width="200" alt="image" src="https://github.com/user-attachments/assets/81328334-a48a-476b-aacd-0e322b102be0" />
<img width="200" alt="image" src="https://github.com/user-attachments/assets/0e5ea7a3-f63a-4158-93c1-5826539180fa" />


There is also an issue with the text at smaller screen widths.


<img width="200" alt="image" src="https://github.com/user-attachments/assets/8ea7d4a5-1274-4890-802e-7c8aa32209c2" />
<img width="200" alt="image" src="https://github.com/user-attachments/assets/4b81d6a8-580c-44fd-81ea-497223a767a9" />

Instead of margins, I would try to use flexbox for alignment, maybe something like this, and you could try positioning that absolutely, maybe like ~40% of the y-axis? This may differ between mobile and desktop, and I encourage you to experiment to get something closer to the design

Warri0r2005 commented:
> <img alt="image" width="400" src="https://private-user-images.githubusercontent.com/119282717/452198021-e6bec37e-47f8-4ac5-8124-ceffe45c8460.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNTA2MTMsIm5iZiI6MTc1MDE1MDMxMywicGF0aCI6Ii8xMTkyODI3MTcvNDUyMTk4MDIxLWU2YmVjMzdlLTQ3ZjgtNGFjNS04MTI0LWNlZmZlNDVjODQ2MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxN1QwODUxNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MDI2NWQzYWIyNzAyMzgyMzk1OGZjZGJmMzM1ZjU1MmIxODBkOGQ1YzUzOWMyYzA3N2I0MmU2MGQwNWE1NWQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.R-kCZ6msKrHVA6GL-4EOu1m8bOGFnLyk7vORl58U0Us"> <img alt="image" width="400" src="https://private-user-images.githubusercontent.com/119282717/452197521-091d8481-3eef-4206-8936-c823af51309a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNTA2MTMsIm5iZiI6MTc1MDE1MDMxMywicGF0aCI6Ii8xMTkyODI3MTcvNDUyMTk3NTIxLTA5MWQ4NDgxLTNlZWYtNDIwNi04OTM2LWM4MjNhZjUxMzA5YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxN1QwODUxNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NTA5MjU3ZGYyNTI1Y2EzODRlMTUyYmI4YjExYWNkYTI1MjQyOTZhNjgzNmYyMzE4ZjllMTBhZTNhYWIxZWNhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Ip4wM0Vz8YyQGvvbqbxdiSfMOjkIJqkoQXwB-NkRdyk">
> There are still some discrepancies between this and the figma design. I would double check
> 
> * the font sizes between the mobile, tablet, and figma designs, and how you have it (currently the figma specifies 16px for desktop but you have it as 20px)
> * the "welcome to AUCC" text also seems bigger
> * the gradient overlay also seems to be lighter in the actual design
> 
> <img alt="image" width="200" src="https://private-user-images.githubusercontent.com/119282717/452203253-81328334-a48a-476b-aacd-0e322b102be0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNTA2MTMsIm5iZiI6MTc1MDE1MDMxMywicGF0aCI6Ii8xMTkyODI3MTcvNDUyMjAzMjUzLTgxMzI4MzM0LWE0OGEtNDc2Yi1hYWNkLTBlMzIyYjEwMmJlMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxN1QwODUxNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMTY2ZGUwNDNkYTRiY2ZiNGZlNThkYzA2MzI5OWVhYTc3NjA4MmY4ODc0NmFiMzc3NzE3ZDEzNTdjYTBhMmFjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.f5AYwYeXD3qB27iKyoRrW1bHZpLcSESMz1rNAf2Eyts"> <img alt="image" width="200" src="https://private-user-images.githubusercontent.com/119282717/452203674-0e5ea7a3-f63a-4158-93c1-5826539180fa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNTA2MTMsIm5iZiI6MTc1MDE1MDMxMywicGF0aCI6Ii8xMTkyODI3MTcvNDUyMjAzNjc0LTBlNWVhN2EzLWY2M2EtNDE1OC05M2MxLTU4MjY1MzkxODBmYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxN1QwODUxNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NjVlMzBmOTU5YjRjNWM1YmFmY2Q0NzgzOTM5NGVlMmJjOTEwZDI3ZjllOWE2NDY3NWQ0MzIwNTQ5OTVmYzBkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rnOg7lDFq0m3xR6yKO2okR73I2B1SfFJHHu-gpS5b1k">
> There is also an issue with the text at smaller screen widths.
> 
> <img alt="image" width="200" src="https://private-user-images.githubusercontent.com/119282717/452204073-8ea7d4a5-1274-4890-802e-7c8aa32209c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNTA2MTMsIm5iZiI6MTc1MDE1MDMxMywicGF0aCI6Ii8xMTkyODI3MTcvNDUyMjA0MDczLThlYTdkNGE1LTEyNzQtNDg5MC04MDJlLTdjOGFhMzIyMDljMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxN1QwODUxNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00OTYwNTEzZWE1NTZlOTE1MWY2ZWZkYjlhN2U0NDQ2ZTFhM2U1OWEzODU3YjU4NThkZmEyNDlmNzFhOTViOGMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.p5BrMXx1JJCn3mPXb5reF1B_YbyIWuJt_1HN0-b9LUs"> <img alt="image" width="200" src="https://private-user-images.githubusercontent.com/119282717/452207514-4b81d6a8-580c-44fd-81ea-497223a767a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAxNTA2MTMsIm5iZiI6MTc1MDE1MDMxMywicGF0aCI6Ii8xMTkyODI3MTcvNDUyMjA3NTE0LTRiODFkNmE4LTU4MGMtNDRmZC04MWVhLTQ5NzIyM2E3NjdhOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYxN1QwODUxNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMzY1YzgwY2JiMTA3MTVjMjg2NWUyM2E2NjFmYzNiZTNiZTg4MDliYmE1YmRlNTFmYmJlY2QzYzUzNmE3Y2RhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DXxlYd5ubBgexbqU6VdhBW3Fi7-BLKXI3C5lGbUWEAU">
> Instead of margins, I would try to use flexbox for alignment, maybe something like this, and you could try positioning that absolutely, maybe like ~40% of the y-axis? This may differ between mobile and desktop, and I encourage you to experiment to get something closer to the design
> 
> Do also add the "scroll to explore" arrow, and the svg curve at the bottom as well

I have updated a few of the spacings to flex and updated the text sizes for all the view ports. Did we have a global component for the scroll cue or should i make one?

dyzhuu commented:
> I've updated a few of the spacings to flex and updated the text sizes for all the view ports so should be all good now. Did we have a global component for the scroll cue or should i make one?

We don't have a component for that yet -- I think you should make it a local component

phawatboom reviewed (approved):
Awesome work! Slight discrepancies in the background overlay color, and the member button size and whitewater kayaking text on the medium screen, but other than that, everything looks pretty good to me!

dyzhuu reviewed (approved):
Wow, this is some absolutely fantastic work! LGTM 🔥🔥🔥
